### ISSUE TITLE
azurerm_function_app & azurerm_app_service missing remote Git source control options

### ISSUE THREAD

**Issue by benc-uk** [terraform-providers/terraform-provider-azurerm#1801]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I am using the **azurerm_function_app** resource and I want to deploy my code to it from GitHub using an external Git source repo URL. This is a standard feature of all Azure App Services including Function Apps. 

You already support this in the **azurerm_app_service** but for some reason have removed/omitted it from the **azurerm_function_app** despite from an Azure ARM perspective they are the same thing.

**EDIT: Correction - you don't appear to support setting this configuration but only reading the values for LocalGit deployment, so my issue applies equally to both resources**

Meaning this is possible with ARM templates to deploy a fully working Function App/App Service but not with Terraform, and IMO makes the **azurerm_function_app** & **azurerm_app_service** resources not terribly useful 

The nested resource is `Microsoft.Web/sites/sourcecontrols` and it accepts `repoUrl` and `branch` properties to deploy from the remote Git URL

This is fully documented here:  
https://docs.microsoft.com/en-us/azure/azure-functions/functions-infrastructure-as-code#create-a-function-app-1

### New or Affected Resource(s)

* azurerm_function_app
* azurerm_app_service 

### Potential Terraform Configuration

Example of what it could look like
```hcl
resource "azurerm_function_app" "foo" {
  name = "bar"
  location = "northeurope"
  app_service_plan_id = "${azurerm_app_service_plan.blahblah.id}"
  source_control {
    repo_url = "https://github.com/benc-uk/serverless-cognitive.git"
    branch = "master"
  }
}
```

### References

* https://docs.microsoft.com/en-us/azure/azure-functions/functions-infrastructure-as-code
* Here is a working ARM template with this feature https://github.com/benc-uk/azure-arm/blob/master/old/paas-other/function-app-withcode/azuredeploy.json

**Comment by benc-uk** [comment]
Here is what the nested source control resource should look like
NOTE. That **isManualIntegration** is set to true when using sourcecontrols in this mode, when using with LocalGit this value will be false

```
{
  "id": "/subscriptions/---/resourceGroups/Demo.ServerlessVision/providers/Microsoft.Web/sites/demofunctionapp/sourcecontrols/web",
  "name": "demofunctionapp",
  "type": "Microsoft.Web/sites/sourcecontrols",
  "location": "North Europe",
  "properties": {
    "repoUrl": "https://github.com/benc-uk/serverless-cognitive.git",
    "branch": "master",
    "isManualIntegration": true,
    "deploymentRollbackEnabled": false,
    "isMercurial": false,
    "provisioningState": "Succeeded"
  }
}
```

**Comment by benc-uk** [comment]
I've checked the Go SDK for Azure and setting/updating this is covered by the SDK
https://github.com/Azure/azure-sdk-for-go/tree/master/services/web/mgmt/2018-02-01/web

e.g. `AppClient.CreateOrUpdateSourceControl()`

**Comment by metacpp** [comment]
Thanks @benc-uk for raising this request. We have #1104 to track the effort to support Git source control as pipeline. It requires webhook to be installed on repo, so need to use personal access token to finish the work. We will share more information in the upcoming PR.

**Comment by benc-uk** [comment]
Thanks

The feature of App Service I'm referring to doesn't require webhooks or PAT tokens. It's a simple git pull from the public URL and Kudu runs the build.

This is not something new, most Azure users have been using it for many years

All that is required is a simple nested ARM resource, with three properties `repoUrl`, `branch` and `isManualIntegration`. That's it. Kudu and Service will do the rest. 
No need to over complicate things

**Comment by benc-uk** [comment]
I have a working example of using the Go SDK for Azure to do what I am talking about

https://github.com/benc-uk/go-azure/blob/2484d00ab5293a28f77eff073c0bb6ed9214d7d1/src/webapp/main.go#L142

**Comment by katbyte** [comment]
Closed as a duplicate of #1104
